### DISCUSSION TITLE
Invalidate Cache and regeneration

### DISCUSSION THREAD

**Discussion by Pedromdsn** [vercel/next.js#33961] [Help] [answered]
### Describe the feature you'd like to request

If I have a blog page with an ISR strategy with a 30 minute revalidate when I add a post I need to wait until the time runs out for that post to appear on the main page.

### Describe the solution you'd like

Could have a conditional revalidate strategy, when a function is called it would revalidate a certain page.

```js
revalidate("news");
```

That would invalidate the page that the path was news and start a revalidation, so when I go back to the home page after submitting the post, if it didn't give any error, it would already show the updated page.

### Describe alternatives you've considered

If the other solution doesn't work, I also believe that we could put dependencies (similar to useEffect) in the ISR.

```js
export async function getStaticProps() {
  const res = await fetch('https://.../posts')
  const posts = await res.json()

  return {
    props: {
      posts,
    },
    revalidate: 10,
    revalidateif: [{req: ["post"], route: "api/posts" }]
  }
}
```

If the api/posts route was called by the post method, the page would be revalidated.

**Comment by leerob** [comment, marked as answer]
I'm incredibly excited to share that Next.js 12.1 was just released, which includes **on-demand Incremental Static Regeneration**! We know many of you have been looking forward to this for a while, including myself, and I'm so happy it is here. 

- [Learn about Next.js 12.1](https://nextjs.org/blog/next-12-1)
- [Watch a demo of on-demand ISR](https://youtu.be/BGexHR1tuOA)
- [View another demo and give us your feedback](https://on-demand-isr.vercel.app/)
- [Read the documentation](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation-beta)
